Chessington.GameEngine/Helpers.cs

```diff
@@ -4,30 +4,56 @@
 
 namespace Chessington.GameEngine
 {
-	public class Helpers
+    public class Helpers
     {
-        public List<Direction> AddMovesAndUpdateDirections(Board board, List<Direction> directions, Square currentSquare, int i, List<Square> moves, Player player)
+        public void AddMoves(
+            Board board, 
+            List<Direction> directions, 
+            Square currentSquare, 
+            int spacesAway, 
+            List<Square> moves, 
+            Player player
+            )
         {
-            var toRemove = new List<Direction>();
+            var newDirections = new List<Direction>();
             foreach (var direction in directions)
             {
-                var potentialMove = Square.At(currentSquare.Row + i * direction.RowOffset,
-                    currentSquare.Col + i * direction.ColOffset);
-                if (board.GetPiece(potentialMove) != null)
+                var potentialMove = Square.At(
+                    currentSquare.Row + spacesAway * direction.RowOffset,
+                    currentSquare.Col + spacesAway * direction.ColOffset
+                    );
+                var pieceOnNewSquare = board.GetPiece(potentialMove);
+
+                if (pieceOnNewSquare == null || pieceOnNewSquare.Player != player)
                 {
-                    toRemove.Add(direction);
-                    if (board.GetPiece(potentialMove).Player != player)
-                    {
-                        moves.AddIfOnBoard(potentialMove);
-                    }
+                    moves.AddIfOnBoard(potentialMove);
                 }
-                else
+            }
+        }
+
+        public List<Direction> UpdateValidDirections(
+            Board board,
+            List<Direction> directions,
+            Square currentSquare,
+            int spacesAway
+        )
+        {
+            var newDirections = new List<Direction>();
+            foreach (var direction in directions)
```

Comment:
You can probably simplify this with `Where`. Something like:
```
return directions.Where(
  direction => SpaceInDirectionIsAvailable(direction, board, currentSquare, spacesAway)
)
```
where `SpaceInDirectionIsAvailable` is a private helper method which calculates the potential move and checks if the space is available. Does that make sense? 🙂